IdentityCore/src/cache/MSIDToken.h

```diff
@@ -22,13 +22,13 @@
 // THE SOFTWARE.
 
 #import <Foundation/Foundation.h>
-#import "MSIDJsonObject.h"
+#import "MSIDJsonSerializable.h"
 #import "MSIDTokenResponse.h"
 #import "MSIDRequestParameters.h"
 #import "MSIDClientInfo.h"
 #import "MSIDTokenType.h"
 
-@interface MSIDToken : MSIDJsonObject <NSCopying, NSSecureCoding>
+@interface MSIDToken : NSObject <NSCopying, NSSecureCoding, MSIDJsonSerializable>
```

Comment:
So, to make sure I understand correctly, the reason for not using MSIDJsonObject are those properties, which need to be not NSString:
- cached at
- expires on
- clientInfo
- authority

I'm wondering, how much performance/code maintainability difference is there? All those properties are readonly, so do we actually have that much complexity/performance hit around syncing them? 

Not saying any of the solutions are right or wrong, just making sure we evaluated all the data we have :)

Reply:
1. MSIDJsonObject easily allows to store only 5 properties (idToken, familyId, resource, clientId, additionalServerInfo). Everything else requires adding either temporary variables or extra logic to make it work with MSIDJsonObject, in particular: 
- authority/environment
- scopes
- clientInfo
- expiresOn
- token/token type

2. Adding temporary variables to handle those properties requires us to synchronize them with _json dictionary, that leads to extra complexity and possible bugs.

3. Despite majority of properties are readonly, we still can serialize/deserialize from NSData, so we need to write/read them. Also developer can always use KVC to write 'readonly' properties; current implementation will be still working with KVC, when MSIDJsonObject will not.

4. If we decide to make properties writeable, with current implementation it can be done very easily, when with MSIDJsonObject it will be pretty hard and will bring additional complexity.

5. In general, MSIDJsonSerializable is more flexible. If we have a class Animal that contains some logic and we need a class Dog that should extend Animal and be json serializable/deserializable, we can easily do it with current approach when  MSIDJsonObject doesn't allow to do that.